main.js

```diff
@@ -108,8 +108,107 @@ function funZero(newArray){
 }
 console.log(funZero(newMas));
 
+// дз 4
+// задание 5
+function sum(a) {
+    return function(b) {
+        return a + b;
+    };
+}
+console.log(sum(5)(2));
+// задание 6
+const arrColors = ['magenta', 'cyan', 'firebrick', 'springgreen', 'skyblue'];
+const textFirst = document.getElementById("text1");
+const textSecond= document.getElementById("text2");
+const textThird = document.getElementById("text3");
 
+function changeColor() {
+    let colorIndex = 0;
+    return function (){
+        this.style.color = arrColors[colorIndex];
+        colorIndex++;
+        if (colorIndex === arrColors.length){
+            colorIndex = 0;
+        }
+    }
+}
+textFirst.addEventListener("click", changeColor());
+textSecond.addEventListener("click", changeColor());
+textThird.addEventListener("click", changeColor());
 
+// ДЗ 5
+// задание 5
+function dateFormatting(date) {
+    const dateReg = /^\d{4}-\d{2}-\d{2}$/;
+    if (dateReg.test(date)) {
+        date = date.split('-')
+            .reverse()
+            .join('.');
+        return date;
+    } else {
+        return 'enter correct date format';
+    }
+}
+console.log(dateFormatting())
+
+// Задание 6
+const data = [
+    {
+        country: 'Russia',
+        city: 'Saint Petersburg',
+        hotel: 'Hotel Leopold',
+    },
+    {
+        country: 'Spain',
+        city: 'Santa Cruz de Tenerife',
+        hotel: 'Apartment Sunshine',
+    },
+    {
+        country: 'Slowakia',
+        city: 'Vysokie Tatry',
+        hotel: 'Villa Kunerad',
+    },
+    {
+        country: 'Germany',
+        city: 'Berlin',
+        hotel: 'Hostel Friendship',
+    },
+    {
+        country: 'Indonesia',
+        city: 'Bali',
+        hotel: 'Ubud Bali Resort&SPA',
+    },
+    {
+        country: 'Netherlands',
+        city: 'Rotterdam',
+        hotel: 'King Kong Hostel',
+    },
+    {
+        country: 'Marocco',
+        city: 'Ourika',
+        hotel: 'Rokoko Hotel',
+    },
+    {
+        country: 'Germany',
+        city: 'Berlin',
+        hotel: 'Hotel Rehberge Berlin Mitte',
+    },
+];
+const resultOfSearch = [];
+function search(str) {
+    for (let j = 0; j < data.length; j++) {
+        const dataValues = Object.values(data[j]);
+        if (dataValues.includes(str)) {
+            resultOfSearch.push(dataValues.join(' '));
+        }
+    }
+    if (resultOfSearch.length !== 0) {
+        return resultOfSearch;
+    } else {
+        throw `Совпадений по запросу ${str} нет`;
```

Comment:
Your exception calls an error into the browser console, so, user can see it only when he/she will open the inspect, it's not a good UX(user experience) at all - you can use try/catch block or throw a specific exception

Reply:
it's not a HW case , but read it for the future